activation/poet.go

```diff
@@ -212,6 +208,25 @@ func (c *HTTPPoetClient) CertifierInfo(ctx context.Context) (*url.URL, []byte, e
 	return url, certifierInfo.Pubkey, nil
 }
 
+func (c *HTTPPoetClient) verifyPhaseShiftConfiguration(ctx context.Context) error {
+	if c.fetchedPhaseShift != 0 {
+		return nil
+	}
+
+	resp, err := c.info(ctx)
+	if err != nil {
+		return err
+	} else if resp.PhaseShift.AsDuration() != c.expectedPhaseShift {
+		c.logger.Panic("verifying poet service configuration",
+			zap.Duration("server: phase shift", resp.PhaseShift.AsDuration()),
+			zap.Duration("config: phase shift", c.expectedPhaseShift),
+		)
+	}
```

Comment:
I think this should have a unit test verifying this behaviour when calling `Submit`. 

Hint: if we instead return an error here we and panic at a higher level (caller of `Submit` and `NewHttpPoetClient`) it will be easier to write the test 🙂 

```suggestion
	resp, err := c.info(ctx)
	if err != nil {
		return err
	}
	if resp.PhaseShift.AsDuration() != c.expectedPhaseShift {
		return ErrIncompatibilePhaseShift
	}
```